tests/communication/test_decorator.py

```diff
@@ -0,0 +1,91 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+from concurrent.futures import ThreadPoolExecutor, as_completed
+import os
+import redis
+import subprocess
+import sys
+import time
+import threading
+import unittest
+
+from maro.communication import Proxy, SessionMessage, dist
+
+
+def proxy_generator(component_type):
+    if component_type == "receiver":
+        proxy = Proxy(group_name="proxy_unit_test",
+                      component_type="receiver",
+                      expected_peers={"sender": 1},
+                      log_enable=False)
+    elif component_type == "sender":
+        proxy = Proxy(group_name="proxy_unit_test",
+                      component_type="sender",
+                      expected_peers={"receiver": 1},
+                      log_enable=False)
+    return proxy
+
+def handler_function(that, proxy, message):
+    replied_payload = message.payload + 1
+    proxy.reply(message, payload=replied_payload)
+    sys.exit(0)
+
+def decorator_start(proxy, handler_dict):
+    @dist(proxy, handler_dict)
+    class Receiver:
+        def __init__(self):
+            pass
+    
+    decorator = Receiver()
+    decorator.launch()
+
+@unittest.skipUnless(os.environ.get("test_with_redis", False), "require redis")
+class TestDecorator(unittest.TestCase):
+    @classmethod
+    def setUpClass(cls):
+        print(f"The dist decorator unit test start!")
+        # Prepare Redis.
+        try:
```

Comment:
change it to create a new redis-server with a random port, and at the teardown stage close this redis-server with its pid.

Reply:
also for others parts